tools/bcdb_mock/pkg/escrow/escrow.go

```diff
@@ -0,0 +1,332 @@
+		reservation  workloads.Reservation
+		responseChan chan reservationRegisterJobResponse
+	}
+
+	reservationRegisterJobResponse struct {
+		data []types.EscrowDetail
+		err  error
+	}
+)
+
+const (
+	// interval between every check of active escrow accounts
+	balanceCheckInterval = time.Minute * 1
+)
+
+// New creates a new escrow object and fetches all addresses for the escrow wallet
+func New(wallet *stellar.Wallet, db *mongo.Database) *Escrow {
+	jobChannel := make(chan reservationRegisterJob)
+	deployChannel := make(chan schema.ID)
+	cancelChannel := make(chan schema.ID)
+
+	return &Escrow{
+		wallet:             wallet,
+		db:                 db,
+		nodeAPI:            &directory.NodeAPI{},
+		farmAPI:            &directory.FarmAPI{},
+		reservationChannel: jobChannel,
+		deployedChannel:    deployChannel,
+		cancelledChannel:   cancelChannel,
+	}
+}
+
+// Run the escrow until the context is done
+func (e *Escrow) Run(ctx context.Context) error {
+	ticker := time.NewTicker(balanceCheckInterval)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ctx.Done():
+			return nil
+		case <-ticker.C:
+			err := e.checkReservations(ctx)
+			if err != nil {
+				log.Error().Msgf("could not check reservations: %s", err)
+			}
+		case job := <-e.reservationChannel:
+			details, err := e.processReservation(ctx, job.reservation)
+			if err != nil {
+				log.Error().Msgf("could not check reservations: %s", err)
+			}
+			job.responseChan <- reservationRegisterJobResponse{
+				err:  err,
+				data: details,
+			}
+		case id := <-e.deployedChannel:
+			err := e.payoutFarmers(ctx, id)
+			if err != nil {
+				log.Error().Msgf("could not payout farmers: %s", err)
+			}
+		case id := <-e.cancelledChannel:
+			err := e.refundClients(ctx, id)
+			if err != nil {
+				log.Error().Msgf("could not refund clients: %s", err)
+			}
+		}
+	}
+}
+
+// checkReservations checks all the active reservations and marks those who are funded.
+// if a reservation is funded then it will mark this reservation as to DEPLOY.
+// if its underfunded it will throw an error.
+func (e *Escrow) checkReservations(ctx context.Context) error {
+	log.Debug().Msg("scanning active ecrow accounts balance")
+	// load active escrows
+	reservationEscrows, err := types.GetAllActiveReservationPaymentInfos(ctx, e.db)
+	if err != nil {
+		return errors.Wrap(err, "failed to load active reservations from escrow")
+	}
+	for _, escrowInfo := range reservationEscrows {
+		allPaid := true
+		for _, escrowAccount := range escrowInfo.Infos {
+			balance, _, err := e.wallet.GetBalance(escrowAccount.EscrowAddress, escrowInfo.ReservationID)
+			if err != nil {
+				allPaid = false
+				return errors.Wrap(err, "failed to verify escrow account balance")
+			}
+			if balance < escrowAccount.TotalAmount {
+				allPaid = false
+				return errors.Wrapf(err, "escrow account %s for reservation id %d is not funded yet", escrowAccount.EscrowAddress, escrowInfo.ReservationID)
+			}
+		}
+		if allPaid {
```

Comment:
This condition doesn't make sense because if allPaid != true, the function would return already and never reach this block. hence the check can be dropped